packages/vm/src/instance.rs

```diff
@@ -391,6 +430,53 @@ mod tests {
         assert!(instance.required_features().contains("water"));
     }
 
+    #[test]
+    fn extra_imports_get_added() {
+        let wasm = wat::parse_str(
+            r#"(module
+            (import "foo" "bar" (func $bar))
+            (func (export "main") (call $bar))
+            )"#,
+        )
+        .unwrap();
+
+        let backend = mock_backend(&[]);
+        let (instance_options, memory_limit) = mock_instance_options();
+        let module = compile(&wasm, memory_limit, &[]).unwrap();
+
+        #[derive(wasmer::WasmerEnv, Clone)]
+        struct MyEnv {
+            // This can be mutated across threads safely. We initialize it as `false`
+            // and let our imported fn switch it to `true` to confirm it works.
+            called: Arc<AtomicBool>,
+        }
+
+        let my_env = MyEnv {
+            called: Arc::new(AtomicBool::new(false)),
+        };
+
+        let fun = Function::new_native_with_env(module.store(), my_env.clone(), |env: &MyEnv| {
+            env.called.store(true, Ordering::Relaxed);
+        });
```

Comment:
This now has a completely different env than the internal Environment instance from https://github.com/CosmWasm/cosmwasm/blob/v0.16.2/packages/vm/src/instance.rs#L77.

Do you think this can become an issue?

Reply:
I assumed each function can very well have a separate env. I don't see anything in here that would indicate otherwise: https://docs.rs/wasmer/1.0.0-rc1/wasmer/trait.WasmerEnv.html

I could do a little bit of extra testing to make sure, though.